libp2p/Common.h

```diff
@@ -254,26 +254,24 @@ class DeadlineOps
     public:
         DeadlineOp(ba::io_service& _io, unsigned _msInFuture, std::function<void(boost::system::error_code const&)> const& _f): m_timer(new ba::deadline_timer(_io)) { m_timer->expires_from_now(boost::posix_time::milliseconds(_msInFuture)); m_timer->async_wait(_f); }
         ~DeadlineOp() { if (m_timer) m_timer->cancel(); }
-        
-        DeadlineOp(DeadlineOp&& _s): m_timer(_s.m_timer.release()) {}
-        DeadlineOp& operator=(DeadlineOp&& _s)
-        {
-            assert(&_s != this);
 
-            m_timer.reset(_s.m_timer.release());
+        DeadlineOp(DeadlineOp&& _s) noexcept : m_timer(std::move(_s.m_timer)) {}
```

Comment:
Why do we need noexcept here?

Reply:
Move constructors are much more powerful with `noexcept`. Clang-tidy suggests this change. https://en.cppreference.com/w/cpp/utility/move_if_noexcept.